### PR TITLE
PLT-1307 Make proper async calls when switching channels

### PR DESCRIPTION
**Pull request by jwilander**
Also moved counting channel unreads to the channel store where it belongs.

### CODE REVIEW

File: web/react/components/sidebar.jsx [hunk 1]
```diff
@@ -157,6 +125,7 @@ export default class Sidebar extends React.Component {
             privateChannels,
             visibleDirectChannels,
             hiddenDirectChannelCount,
+            unreadCounts: JSON.parse(JSON.stringify(ChannelStore.getUnreadCounts())),
```
Comment (crspeller) [inline web/react/components/sidebar.jsx]: Maybe this should be done in the store and not here.

File: web/react/components/sidebar.jsx [hunk 2]
```diff
@@ -204,10 +171,9 @@ export default class Sidebar extends React.Component {
             windowHeight: Utils.windowHeight()
         });
     }
-    updateScrollbar() {
-    }
     onChange() {
-        this.setState(this.getStateFromStores());
+        const newState = this.getStateFromStores();
+        this.setState(newState);
```
Comment (crspeller) [inline web/react/components/sidebar.jsx]: Why the change? The assignment does not copy. Just want to make sure that this wasn't trying to do that.

Comment (jwilander) [inline web/react/components/sidebar.jsx]: No reason, I was actually logging it and forgot to change it back

File: web/react/components/sidebar.jsx (hunk 1 above)
Comment (jwilander) [inline web/react/components/sidebar.jsx]: Yeah, I think that's probably better and I think that anytime the store returns anything it should be a deep copy but I think I'd like to talk about this in the developer meeting first. Right now I'd like to leave it as is just so the stores are consistent. 0/5 though, if you feel strongly about it I can move it to the store

Comment (crspeller) [inline web/react/components/sidebar.jsx]: Yes I think your right. Lets merge this and talk about it in the dev meeting today.

**Comment by crspeller** [comment]
+1

**Comment by hmhealey** [comment]
+1

**Comment by ninja-** [comment]
@jwilander :exclamation: this broke the app and it doesn't load

> Uncaught TypeError: _channel_store2.default.getUnreadCounts is not a function

**Comment by ninja-** [comment]
nevermind...I think it was some caching problem on dev machine
